core/src/test/java/org/apache/iceberg/avro/TestBuildAvroProjection.java

```diff
@@ -401,4 +402,32 @@ public void projectMapWithLessFieldInValueSchema() {
         .as("Unexpected value ID discovered on the projected map schema")
         .isEqualTo(1);
   }
+
+  @Test
+  public void projectVariantSchemaUnchanged() {
+    final Type icebergType = Types.VariantType.get();
+
+    final org.apache.avro.Schema expected =
+        SchemaBuilder.record("variant")
+            .namespace("unit.test")
+            .fields()
+            .name("metadata")
```

Comment:
I don't think that variant `value` and `metadata` columns are assigned IDs. Instead, the variant field is assigned an ID and these are left alone.

Reply:
You are right. variant `value` and `metadata` should not be assigned IDs.